lib/palettes.cpp

```diff
@@ -7703,16 +221,15 @@ void PaletteState::Init()
       MaxTextureSize = std::min(MaxTextureSize, 4096);
       {
          std::lock_guard<std::mutex> lk{init_mtx};
-         Init_Palettes();
       }
 
-      GLuint paletteTexIds[Num_RGB_Palettes][2];
+      vector<array<GLuint, 2>> paletteTexIds(Palettes->MAX_PALETTES);
```

Comment:
Why this? 🤪  The number of palettes cannot be changed after init anyway, because there is no re-init...

Reply:
Initialization of `paletteTexIds` using `NumPalettes()` throws a CI error because the dynamic allocation isn't compatible with `c++99`. There is no re-init atm but I think most (all?) of `PaletteState::Init()` should go into `Palette` anyways; that way you can add palettes at any point, eg in `pyglvis`.

Reply:
I understand that a C array is a problem, but `std::vector` should be ok. Secondly, where is `c++99`? MFEM requires C++ 11 🤪 .
It is batched here in a way, but it should not be a big performance hit to allocate the textures separately in `Palette` directly, I believe, and it would be much cleaner 😉 .

Reply:
https://github.com/GLVis/glvis/actions/runs/11693833399/job/32566220453#step:17:27 is what I was referring to, but looking back it was an array not a vector back then 🤔 maybe I'm misremembering 😅. 

Regardless, yeah I'll just try to move this and other relevant things into `Palette`; it'll be much cleaner and would also be a nice way to utilize the alpha channel in the palette file. The only thing I'm not sure of is currently the alpha adjustment <kbd>k</kbd> / <kbd>K</kbd> is a global operation. If a colormap has a custom alpha channel and the user makes an adjustment, do we override/ignore/multiply?

Reply:
Yeah, it is about a C array, where you can use only `constexpr`. It just mentions variable-length arrays from C99 as a suggestion, but that is a different story.
That is a clean approach to move things to `Palette`. With alpha, I would multiply, that is the most consistent behavior, I would say 🤔 .

Reply:
I think we can resolve this?